sktime/forecasting/compose/tests/test_fallback.py

```diff
@@ -0,0 +1,320 @@
+            ),
+            ("forecaster4", PolynomialTrendForecaster()),
+        ]
+    )
+    forecaster.fit(y=y, fh=[1, 2, 3])
+    y_pred_actual = forecaster.predict()
+    y_pred_expected = ensemble_forecaster_3.predict()
+    name = forecaster.current_name
+    assert name == "ensemble3"
+    pd.testing.assert_series_equal(y_pred_expected, y_pred_actual)
+
+
+def test_fallbackforecaster_fails_fit_twice():
+    """First two FallbackForecasters fail at fit step, third forecaster succeeds"""
+    y = make_forecasting_problem(random_state=42)
+    ensemble_forecaster_1 = EnsembleForecaster(
+        [
+            ("trend", PolynomialTrendForecaster()),
+            ("fails", DummyForecaster(raise_at="fit")),
+        ]
+    )
+    ensemble_forecaster_2 = EnsembleForecaster(
+        [
+            ("trend", PolynomialTrendForecaster()),
+            ("fails", DummyForecaster(raise_at="fit")),
+        ]
+    )
+    ensemble_forecaster_3 = EnsembleForecaster(
+        [
+            ("trend", PolynomialTrendForecaster()),
+            ("naive", NaiveForecaster()),
+        ]
+    )
+    forecaster = FallbackForecaster(
+        [
+            (
+                "ensemble1",
+                ensemble_forecaster_1,
+            ),
+            (
+                "ensemble2",
+                ensemble_forecaster_2,
+            ),
+            (
+                "ensemble3",
+                ensemble_forecaster_3,
+            ),
+            ("forecaster4", PolynomialTrendForecaster()),
+        ]
+    )
+    forecaster.fit(y=y, fh=[1, 2, 3])
+    y_pred_actual = forecaster.predict()
+    y_pred_expected = ensemble_forecaster_3.predict()
+    name = forecaster.current_name
+    assert name == "ensemble3"
+    pd.testing.assert_series_equal(y_pred_expected, y_pred_actual)
+
+
+def test_all_forecasters_fail1():
+    """All forecasters fail; predict and fit"""
+    # Start with negative time series, Theta model will fail here
+    y = make_forecasting_problem(random_state=42)
+    forecaster1 = ("raise_predict1", DummyForecaster(raise_at="predict"))
+    forecaster2 = ("raise_fit1", DummyForecaster(raise_at="fit"))
+    forecaster3 = ("raise_fit2", DummyForecaster(raise_at="fit"))
+    forecaster = FallbackForecaster([forecaster1, forecaster2, forecaster3])
+    with pytest.raises(RuntimeError):
+        forecaster.fit(y=y, fh=[1, 2, 3])
+        forecaster.predict()
+
+
+def test_all_forecasters_fail2():
+    """All forecasters fail at fit step"""
+    # Start with negative time series, Theta model will fail here
+    y = make_forecasting_problem(random_state=42)
+    forecaster1 = ("raise_fit1", DummyForecaster(raise_at="fit"))
+    forecaster2 = ("raise_fit2", DummyForecaster(raise_at="fit"))
+    forecaster3 = ("raise_fit3", DummyForecaster(raise_at="fit"))
+    forecaster = FallbackForecaster([forecaster1, forecaster2, forecaster3])
+    with pytest.raises(RuntimeError):
+        forecaster.fit(y=y, fh=[1, 2, 3])
+        forecaster.predict()
```

Comment:
Is this line reachable, as everything fails during `fit` stage?

Reply:
You're correct, this line can be removed.